cpp/tests/io/csv/csv_test.cu

```diff
@@ -312,443 +312,481 @@ TEST(gdf_csv_test, MortPerf)
-
-	{
-		csv_read_arg args{};
-		args.input_data_form = gdf_csv_input_form::FILE_PATH;
-		args.filepath_or_buffer = fname;
-		args.num_cols = std::extent<decltype(names)>::value;
-		args.names = names;
-		args.dtype = types;
-		args.delimiter = ',';
-		args.lineterminator = '\n';
-		args.skip_blank_lines = true;
-		args.header = -1;
-		args.comment = '#';
-		args.nrows = -1;
-		EXPECT_EQ( read_csv(&args), GDF_SUCCESS );
-
-		EXPECT_EQ( args.num_cols_out, args.num_cols );
-		ASSERT_EQ( args.data[0]->dtype, GDF_INT32 );
-
-		auto ACol = gdf_host_column<int32_t>(args.data[0]);
-		EXPECT_THAT( ACol.hostdata(), ::testing::ElementsAre(1, 3, 4, 5, 8, 9) );
-	}
+    const char* fname = "/tmp/CsvSkiprowsNrows.csv";
+    const char* names[] = { "A" };
+    const char* types[] = { "int32" };
+
+    std::ofstream outfile(fname, std::ofstream::out);
+    outfile << "1\n#blank\n3\n4\n5\n#blank\n\n\n8\n9\n";
+    outfile.close();
+    ASSERT_TRUE( checkFile(fname) );
+
+    {
+        csv_read_arg args{};
+        args.input_data_form = gdf_csv_input_form::FILE_PATH;
+        args.filepath_or_buffer = fname;
+        args.num_cols = std::extent<decltype(names)>::value;
+        args.names = names;
+        args.dtype = types;
+        args.delimiter = ',';
+        args.lineterminator = '\n';
+        args.skip_blank_lines = true;
+        args.header = -1;
+        args.comment = '#';
+        args.nrows = -1;
+        EXPECT_EQ( read_csv(&args), GDF_SUCCESS );
+
+        EXPECT_EQ( args.num_cols_out, args.num_cols );
+        ASSERT_EQ( args.data[0]->dtype, GDF_INT32 );
+
+        auto ACol = gdf_host_column<int32_t>(args.data[0]);
+        EXPECT_THAT( ACol.hostdata(), ::testing::ElementsAre(1, 3, 4, 5, 8, 9) );
+    }
+}
+
+TEST(gdf_csv_test, Writer)
+{
+    const char* fname	= "/tmp/CsvWriteTest.csv";
+    const char* names[] = { "integer", "float", "string" };
+    const char* types[] = { "int32", "float32", "str" };
+
+    std::ofstream outfile(fname, std::ofstream::out);
+    outfile << "1,1.0,one" << '\n';
+    outfile << "2,2.25,two" << '\n';
+    outfile << "3,3.50,three" << '\n';
+    outfile << "4,4.75,four" << '\n';
+    outfile << "5,5.0,five" << '\n';
+    outfile.close();
+
+    csv_read_arg rargs{};
+    rargs.input_data_form = gdf_csv_input_form::FILE_PATH;
+    rargs.filepath_or_buffer = fname;
+    rargs.num_cols = std::extent<decltype(names)>::value;
+    rargs.names = names;
+    rargs.dtype = types;
+    rargs.decimal = '.';
+    rargs.delimiter = ',';
+    rargs.lineterminator = '\n';
+    rargs.skip_blank_lines = true;
+    rargs.header = -1;
+    rargs.nrows = -1;
+    EXPECT_EQ( read_csv(&rargs), GDF_SUCCESS );
+
+    csv_write_arg wargs{};
+    wargs.data = rargs.data;  // columns from reader above
+    wargs.filepath = "/tmp/CsvWriteTestOut.csv";
+    wargs.num_cols = std::extent<decltype(names)>::value;
+    wargs.delimiter = ',';
+    wargs.line_terminator = "\n";
+
+    EXPECT_EQ( write_csv(&wargs), GDF_SUCCESS );
```

Comment:
I don't think this test is sufficient. You should check that the written CSV file is correct.

Reply:
Yes, forgot that part :-)
Verification code added and shows up below.